### DISCUSSION TITLE
New Signal Proposal: Sweep Wave

### DISCUSSION THREAD

**Discussion by OwenDavid** [Hopsan/hopsan#2046] [Show and tell]
# [SignalSweepWave.zip](https://github.com/Hopsan/hopsan/files/5801818/SignalSweepWave.zip)

Attached is code and icon for a proposed new signal. This is a swept sine wave with a frequency that rises over time.

The input parameters match the component in Simster such that the user sets the initial frequency and the rate of frequency change per second. The code is based upon the Sine wave signal component from the default library.

The functionality can also be achieved using two existing signal components, i.e. modulating the sine wave frequency input with another signal.

I propose that a dedicated signal may be more intuitive to use and should be included as it is found as standard in other software (Simster, Amesim). Similarly, a user may not necessarily understand the other signal components well enough to achieve the same output by combining signal components. *This was my personal experience*

We can see the icon and default output in the screenshot below.

![image](https://user-images.githubusercontent.com/66214518/104310016-54a71200-54d3-11eb-9add-eac062357dee.png)

**Comment by robbr48** [comment]
[SignalSweepWave_v2.zip](https://github.com/Hopsan/hopsan/files/5821233/SignalSweepWave_v2.zip)

Looks great! I just simplified your code a little. Most importantly, you don't need to check whether or not a signal port is connected, that feature is already built into Hopsan. You can just use the pointer as it is. It is only necessary if the connection changes the behavior of the equations, as in the triggered ramp.
